scripts/install_third_party_libs_test.py

```diff
@@ -95,9 +94,11 @@ def mock_ensure_directory_exists(unused_path):
             common, 'ensure_directory_exists', mock_ensure_directory_exists)
 
     def test_tweak_yarn_executable(self):
+        call_counter = 0
```

Comment:
> Use CallCounter in test_utils.py (and possibly self.swap to swap a function with its CallCounter'd-equivalent) instead, in the place where you need to track this.

Any reason not to use the existing counter functionality?

Reply:
The existing `CallCounter` class is used to check how many times a particular function is called and requires the method to be defined before we initialize a CallCounter object. However in this case, the output of the method depends on the number of calls previously made to the method. The above functionality can be achieved by using the CallCounter class but would require a whole lot of unnecessary code when we could simply use a local variable.

Reply:
I think you should treat this like you did [here](https://github.com/oppia/oppia/pull/16231#discussion_r1002667338), then -- or, alternatively, have two mock functions mock_is_file_always_return_false and mock_is_file_always_return_true or something like that. Then swap the right thing in the right place and use CallCounter for each.

Alternatively you could return True for a 'valid_filename' and False for an 'invalid_filename'.

But having lots of complex logic in a mock method, especially if that logic is not likely to happen in general, is an antipattern. Don't do it.

Reply:
I have now split the test into two. PTAL. Thanks!